keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -313,6 +319,52 @@ class TitusClusterHandler(
       false
     }
 
+  // todo eb: capture this heuristic and document it for users
+  private suspend fun ResourceDiff<TitusServerGroup>.disableOtherServerGroupJob(resource: Resource<TitusClusterSpec>, desiredVersion: String): Map<String, Any?> {
+    val current = requireNotNull(current) {
+      "Current server group must not be null when generating a disable job"
+    }
+    val existingServerGroups: Map<String, List<ClouddriverTitusServerGroup>> = getExistingServerGroupsByRegion(resource)
+    val sgInRegion = existingServerGroups.getOrDefault(current.location.region, emptyList()).filterNot { it.disabled }
+
+    if (sgInRegion.size < 2) {
+      log.error("Diff says this is not the only active server group, but now we say otherwise. " +
+        "What is going on? Existing server groups: {}", existingServerGroups)
+      throw ActiveServerGroupsException(resource.id, "No other active server group found to disable.")
+    }
+
+    val (rightImageASGs, wrongImageASGs) = sgInRegion
+      .sortedBy { it.createdTime }
+      .partition { it.image.dockerImageVersion == desiredVersion }
+
+    val sgToDisable = when {
+      wrongImageASGs.isNotEmpty() -> {
+        log.debug("Disabling oldest server group with incorrect docker image version for {}", resource.id)
+        wrongImageASGs.first()
```

Comment:
Any particular reason why we would disable just the oldest and not every server group with the wrong image?

Reply:
Ah, I see @robfletcher asked the same question below and you already answered. Fine to address in an upcoming PR.

Reply:
It feels like an unneeded optimization at this point. I likely won't do it in a future PR unless we get feedback about it